algoliasearch/client.go

```diff
@@ -243,6 +243,92 @@ func (c *client) MultipleQueriesWithRequestOptions(queries []IndexedQuery, strat
 	return
 }
 
+func (c *client) ListClusters() (res []Cluster, err error) {
+	return c.ListClustersWithRequestOptions(nil)
+}
+
+func (c *client) ListClustersWithRequestOptions(opts *RequestOptions) (res []Cluster, err error) {
+	var tmp map[string][]Cluster
+	err = c.request(&tmp, "GET", "/1/clusters", nil, read, opts)
+	res = tmp["clusters"]
```

Comment:
`tmp` is a bit smelly in term of code readiness. Use `res` instead as it's really the JSON response from the engine and use `clusters` as a new name for the returned `[]Cluster` variable.